packages/subproviders/src/subproviders/trezor.ts

```diff
@@ -0,0 +1,160 @@
+import { assert } from '@0x/assert';
+import { addressUtils } from '@0x/utils';
+import EthereumTx = require('ethereumjs-tx');
+import * as _ from 'lodash';
+
+import {
+    PartialTxParams,
+    TrezorConnectResponse,
+    TrezorGetAddressResponsePayload,
+    TrezorResponseErrorPayload,
+    TrezorSignMssgResponsePayload,
+    TrezorSignTxResponsePayload,
+    TrezorSubproviderConfig,
+    WalletSubproviderErrors,
+} from '../types';
+
+import { BaseWalletSubprovider } from './base_wallet_subprovider';
+
+const PRIVATE_KEY_PATH = `m/44'/60'/0'`;
+
+export class TrezorSubprovider extends BaseWalletSubprovider {
+    private readonly _publicKeyPath: string;
+    private _cachedAccounts: string[];
+    private readonly _trezorConnectClientApi: any;
+    /**
+     * Instantiates a TrezorSubprovider. Defaults to private key path set to `44'/60'/0'`.
+     * Must be initialized with trezor-connect API module https://github.com/trezor/connect.
+     * @param TrezorSubprovider config object containing trezor-connect API
+     * @return TrezorSubprovider instance
+     */
+    constructor(config: TrezorSubproviderConfig) {
+        super();
+        this._publicKeyPath = PRIVATE_KEY_PATH;
+        this._cachedAccounts = [];
+        this._trezorConnectClientApi = config.trezorConnectClientApi;
+    }
+    /**
+     * Retrieve a users Trezor account. The accounts are private key path derived, This method
+     * is automatically called when issuing a `eth_accounts` JSON RPC request via your providerEngine
+     * instance.
+     * @return An array of accounts
+     */
+    public async getAccountsAsync(): Promise<string[]> {
+        if (this._cachedAccounts.length) {
+            return this._cachedAccounts;
+        }
+        const accounts: string[] = [];
+        const response: TrezorConnectResponse = await this._trezorConnectClientApi.ethereumGetAddress({
+            path: this._publicKeyPath,
+            showOnTrezor: true,
```

Comment:
If we default this to `false`, it's less cumbersome to fetch multiple addresses. My understanding is that this doesn't actually make the process more secure, it just gives the user a false sense of security.

Reply:
Thoughts on this @MarcZenn?